dynamic_programming/01knapsack_recursive.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:math';
+import 'package:test/test.dart';
+
+int knapSackProblem(
+    int numberOfItems, int capacity, List<int> values, List<int> weights) {
```

Comment:
From what i see you could take numberOfItems from `values.length`
```suggestion
int knapSackProblem(int capacity, List<int> values, List<int> weights,
    [int numberOfItems]) {
  numberOfItems ??= values.length;
```